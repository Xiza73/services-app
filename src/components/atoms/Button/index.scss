.bgh-red {
  &:hover {
    @apply bg-red-50;
  }
}

.bgh-green {
  &:hover {
    @apply bg-green-50;
  }
}

.bgh-blue {
  &:hover {
    @apply bg-blue-50;
  }
}

.text-red {
  @apply text-red-100;

  &:hover {
    @apply text-red-150;
  }
}

.text-blue {
  @apply text-blue-100;

  &:hover {
    @apply text-blue-150;
  }
}

.text-green {
  @apply text-green-100;

  &:hover {
    @apply text-green-150;
  }
}
