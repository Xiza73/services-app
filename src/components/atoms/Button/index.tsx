import React, { MouseEventHandler } from "react";
import "./index.scss";

type Props = {
  text: string;
  color: "blue" | "red" | "green";
  border?: boolean;
  onClick?: MouseEventHandler<HTMLButtonElement>;
};

export default function Button({
  text,
  color,
  border = false,
  onClick,
}: Props) {
  const style = border
    ? `border-2 border-solid bgh-${color}`
    : "border-none hover:scale-125 duration-300";

  return (
    <button
      className={`text-${color} font-semibold px-4 py-1 rounded-md ${style} text-md`}
      onClick={onClick!}
      type={`${onClick ? "button" : "submit"}`}
    >
      {text}
    </button>
  );
}
