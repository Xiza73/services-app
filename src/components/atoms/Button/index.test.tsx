/* eslint-disable no-undef */
// trying to use vitest
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";
import Button from "./index";
import React from "react";

function toJson(component: renderer.ReactTestRenderer) {
  const result = component.toJSON();
  expect(result).toBeDefined();
  expect(result).not.toBeInstanceOf(Array);
  return result as renderer.ReactTestRendererJSON;
}

test("renders button", () => {
  const component = renderer.create(<Button color="blue" text="something" />);

  const tree = toJson(component);

  expect(tree).toMatchSnapshot();
});
