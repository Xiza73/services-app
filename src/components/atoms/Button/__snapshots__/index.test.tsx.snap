// Vitest Snapshot v1

exports[`renders button 1`] = `
<button
  className="text-blue font-semibold px-4 py-1 rounded-md border-none hover:scale-125 duration-300 text-md"
>
  something
</button>
`;
